### DISCUSSION TITLE
How to load only 1 frame from GIF

### DISCUSSION THREAD

**Discussion by ggolda** [SixLabors/ImageSharp#2089] [Ideas] [answered]
Is it possible to load only one (first) frame from a GIF without loading the rest? `SixLabors.ImageSharp.Image.Load<Rgba32>(path);` functions loads all of them and I don't see an option that limits it.

E.g. if GIF image have 600 frames, all of them will be loaded in memory when I only need 1.

If it is not possible - do you have any work arounds how I can do that? Maybe load and immediately discard every frame that I don't need?

Thanks!

**Comment by JimBobSquarePants** [comment, marked as answer]
Yeah. By calling `Image.Decode(...)` and using a `GifDecoder` instance with this property set. https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Formats.Gif.GifDecoder.html#SixLabors_ImageSharp_Formats_Gif_GifDecoder_DecodingMode

**Reply by JimBobSquarePants** [reply]
P.S @ggolda The ideas tag is for feature requests. We don't do Q&A here.
